pyleros/rom.py

```diff
@@ -0,0 +1,139 @@
+from myhdl import block, Signal, intbv, instances, always_seq
+from pyleros.types import IM_BITS
+import sys
+
+
+@block
+def pyleros_im(clk, reset, rd_addr, rd_data, filename = None):
+    """Definition of the instruction memory, or the ROM for
+    pyleros. Reading is synchronous with the rising edge of the
+    clock. Writing is not enabled.
+
+    Arguments (ports):
+        clk: The clock signal
+        reset: The reset signal #Async?
+        rd_addr: IN Read address
+        rd_data: OUT The data at IM address
+
+    Parameters:
+        filename: Name of the file or a list containing the instructions
+
+    """
+    IM_SIZE = 2**IM_BITS
+    IM = [(intbv(0)[16:]) for _ in range(IM_SIZE)]
+
+    # print("\n\n")
+    # print("defined mem")
+    # for i in range(10):
+    #   print(IM[i])
+
+    # Fill up the memory registers
+    define_rom(IM, IM_SIZE, filename)
```

Comment:
In the use case, I would want to(while instantiating the core from rhea) pass it a binary file containing the instructions and then expect it to work as expected. That's why I have done that. 

The processor is designed to be used to function as a peripheral in SoC designs instead of a general purpose processor, so the instruction list, once set, I expect won't change much. Is it still better to go with what you suggested?